samples/extensions/debug_utils/debug_utils.cpp

```diff
@@ -0,0 +1,1135 @@
+			std::vector<glm::vec3> mesh_colors = {
+			    glm::vec3(1.0f, 0.0f, 0.0f),
+			    glm::vec3(0.0f, 1.0f, 0.0f),
+			    glm::vec3(0.0f, 0.0f, 1.0f),
+			};
+			std::vector<glm::vec3> mesh_offsets = {
+			    glm::vec3(-2.5f, 0.0f, 0.0f),
+			    glm::vec3(0.0f, 0.0f, 0.0f),
+			    glm::vec3(2.5f, 0.0f, 0.0f),
+			};
+			for (uint32_t j = 0; j < 3; j++)
+			{
+				push_const_block.object_type = 1;
+				push_const_block.offset      = glm::vec4(mesh_offsets[j], 0.0f);
+				push_const_block.color       = glm::vec4(mesh_colors[j], 0.0f);
+				vkCmdPushConstants(draw_cmd_buffers[i], pipeline_layouts.models, VK_SHADER_STAGE_VERTEX_BIT | VK_SHADER_STAGE_FRAGMENT_BIT, 0, sizeof(push_const_block), &push_const_block);
+				cmd_insert_label(draw_cmd_buffers[i], "Draw sphere", {mesh_colors[j].r, mesh_colors[j].g, mesh_colors[j].b, 1.0f});
+				draw_model(models.scene, draw_cmd_buffers[i]);
+			}
+
+			vkCmdEndRenderPass(draw_cmd_buffers[i]);
+
+			cmd_end_label(draw_cmd_buffers[i]);
+		}
+
+		/*
+			Second render pass: First bloom pass
+		*/
+		if (bloom)
+		{
+			cmd_begin_label(draw_cmd_buffers[i], "Separable bloom filter", {0.5f, 0.76f, 0.34f, 1.0f});
+
+			cmd_begin_label(draw_cmd_buffers[i], "Vertical bloom pass", {0.4f, 0.61f, 0.27f, 1.0f});
+
+			VkClearValue clear_values[2];
+			clear_values[0].color        = {{0.0f, 0.0f, 0.0f, 0.0f}};
+			clear_values[1].depthStencil = {0.0f, 0};
+
+			// Bloom filter
+			VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+			render_pass_begin_info.framebuffer              = filter_pass.framebuffer;
+			render_pass_begin_info.renderPass               = filter_pass.render_pass;
+			render_pass_begin_info.clearValueCount          = 1;
+			render_pass_begin_info.renderArea.extent.width  = filter_pass.width;
+			render_pass_begin_info.renderArea.extent.height = filter_pass.height;
+			render_pass_begin_info.pClearValues             = clear_values;
+
+			vkCmdBeginRenderPass(draw_cmd_buffers[i], &render_pass_begin_info, VK_SUBPASS_CONTENTS_INLINE);
+
+			VkViewport viewport = vkb::initializers::viewport((float) filter_pass.width, (float) filter_pass.height, 0.0f, 1.0f);
+			vkCmdSetViewport(draw_cmd_buffers[i], 0, 1, &viewport);
+
+			VkRect2D scissor = vkb::initializers::rect2D(filter_pass.width, filter_pass.height, 0, 0);
+			vkCmdSetScissor(draw_cmd_buffers[i], 0, 1, &scissor);
+
+			vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layouts.bloom_filter, 0, 1, &descriptor_sets.bloom_filter, 0, NULL);
+
+			vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipelines.bloom[1]);
+			vkCmdDraw(draw_cmd_buffers[i], 3, 1, 0, 0);
+
+			vkCmdEndRenderPass(draw_cmd_buffers[i]);
+
+			cmd_end_label(draw_cmd_buffers[i]);
+		}
+
+		/*
+			Note: Explicit synchronization is not required between the render pass, as this is done implicit via sub pass dependencies
```

Comment:
Consider something as simple as;
> Note: Synchronization between render passes is handled via sub pass dependencies.

This is because taking about 'explicit' and 'implicit' synchronization could be confusing; the synchronization is not really implicit here -- it's just defined elsewhere. Additionally, talking about 'implicit' sub pass dependencies risks confusing the current explicit ones with the the (actually) implicit subpass dependency which will be added by the implementation when custom VK_SUBPASS_EXTERNAL dependencies are not present.